server/src/operators/chunk_operator.rs

```diff
@@ -1438,22 +1439,28 @@ pub fn get_highlights_with_exact_match(
         engine.insert(i, x);
     });
 
+    let mut query_parts_used = HashSet::new();
+
     let mut matched_idxs = split_content
         .iter()
         .enumerate()
-        .filter_map(
-            |(i, split)| match query_parts_split_by_stop_words.contains(split) {
-                true => Some((i, split.clone())),
+        .filter_map(|(i, split)| {
+            match query_parts_split_by_stop_words.contains(&split.to_lowercase()) {
+                true => {
+                    query_parts_used.insert(split.clone());
+                    Some((i, split.clone()))
+                },
                 false => None,
-            },
-        )
+            }
+        })
         .collect_vec();
 
     let exact_matches = matched_idxs.clone();
 
     let results: Vec<usize> = query_parts_split_by_stop_words
         .clone()
         .iter()
+        .filter(|p| !query_parts_used.contains(&p.to_string()))
```

Comment:
whether or not we should do this is debatable, but i think this let's us avoid the search engine giving us results that may overlap with exact matches